Sources/MessagingPushAPN/MessagingPushAPN+PushConfigs.swift

```diff
@@ -45,6 +45,6 @@ extension MessagingPushAPN {
     // Swizzled method for `didFailToRegisterForRemoteNotificationsWithError'
     @objc
     func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
-        messagingPush.deleteDeviceToken()
+        Self.shared.deleteDeviceToken()
```

Comment:
Great additions to the PR description. Thanks! 

From the updated PR description, this is my understanding. 

This bug only happens when: 
* The messaging push module is never initialized in the app, resulting in `messagingPush` instance never being populated. 
* Device token failing to be generated for the host app. 

This seems like a rare bug to encounter with a workaround of initializing the messagingpush module in the app. 

If my understanding is correct, and there is a workaround available, I would appreciate it if we could prioritize writing automated tests in the PR before merging. This will help ensure that the fix is reliable both now and in the future. 

Besides automated testing, I do have another idea to share. 

Could we instead call one of the public-facing API functions like a customer would call? 

```suggestion
        MessagingPushAPN.shared.deleteDeviceToken()
```

If we call the function like a customer would, we can be extra confident that the app would not crash.

Reply:
Done. Please re-review the PR on priority as the customer is expecting a solution from past 2 weeks.